### DISCUSSION TITLE
Specifying a port does not work, always a random port is assigned.

### DISCUSSION THREAD

**Discussion by ClaushIV** [fastapi/fastapi#6133] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app= FastAPI(
    redoc_url= None,
)

app.get("/")
async def root():
    return ("test")
```

### Description

I am trying to run the code above,

`python -m uvicorn main:app --port=8000 --reload`

Which runs the API, that part is OK. However, the port is always assigned randomly and I cannot find a way to fix this.

INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [2432] using StatReload
INFO:     Started server process [2524]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:64692 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:64692 - "GET /openapi.json HTTP/1.1" 200 OK

### Operating System

Windows

### Operating System Details

Windows 11 Home SL

### FastAPI Version

0.68.0

### Python Version

3.10.2

**Comment by Kludex** [comment, marked as answer]
> INFO: Uvicorn running on http://127.0.0.1:8000/ (Press CTRL+C to quit)

Server :point_up:

> INFO: 127.0.0.1:64692 - "GET /docs HTTP/1.1" 200 OK
INFO: 127.0.0.1:64692 - "GET /openapi.json HTTP/1.1" 200 OK

Client ☝️

**Comment by Blue9** [comment]
As @Kludex mentioned, `127.0.0.1:64692` is the client address, and your server is still running on port 8000. If you'd like to learn more you can read about [ephemeral ports](https://en.wikipedia.org/wiki/Ephemeral_port).

**Comment by ClaushIV** [comment]
I will make sure to check it out, thank you very much. 

Well, it appears there is some other issue with my current project. Because initially I thought that the problem was here because of the port problem.

Everything is still the same, however, whatever I add (such as the root function in the example code) does not appear on swagger ui.

When I add another function, such as
```
app.get("/test")
async def test(data: str):
    return {"data": data}
```

This is the output in the console
INFO:     127.0.0.1:65245 - "GET /test HTTP/1.1" 404 Not Found
And when I check it through chrome:
{"detail":"Not Found"}

**Comment by acidjunk** [comment]
Don't you mean

```
@app.get("/test")
```

The decorator is important.

**Comment by acidjunk** [comment]
Probably good to close this ticket when the port question was resolved.

**Comment by ClaushIV** [comment]
> 

just a typo from me, but you're right. the issue is out of topic so I' am closing the topic. thanks anyway.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @ClaushIV  👍
